provers/stark/src/transcript.rs

```diff
@@ -15,18 +15,22 @@ pub trait IsStarkTranscript<F: IsField> {
     fn state(&self) -> [u8; 32];
     fn sample_field_element(&mut self) -> FieldElement<F>;
     fn sample_u64(&mut self, upper_bound: u64) -> u64;
-    fn sample_z_ood(
+    fn sample_z_ood<S: IsSubFieldOf<F>>(
         &mut self,
-        lde_roots_of_unity_coset: &[FieldElement<F>],
-        trace_roots_of_unity: &[FieldElement<F>],
+        lde_roots_of_unity_coset: &[FieldElement<S>],
+        trace_roots_of_unity: &[FieldElement<S>],
     ) -> FieldElement<F>
     where
         FieldElement<F>: Serializable,
     {
         loop {
             let value: FieldElement<F> = self.sample_field_element();
-            if !lde_roots_of_unity_coset.iter().any(|x| x == &value)
-                && !trace_roots_of_unity.iter().any(|x| x == &value)
+            if !lde_roots_of_unity_coset
```

Comment:
Given that roots of unity are real for the cases we have, it would suffice to check if the imaginary part is non-zero to know it is an ood point, avoiding the cloning and extension